WorkflowSwiftUIExperimental/Sources/SwiftUIScreen.swift

```diff
@@ -144,21 +145,31 @@ private final class ModeledHostingController<Model, Content: View>: UIHostingCon
 
         defer { hasLaidOutOnce = true }
 
-        if #available(iOS 16.0, *) {
-            // Handled in initializer, but set it on first layout to resolve a bug where the PCS is
-            // not updated appropriately after the first layout.
-            // UI-5797
-            if !hasLaidOutOnce,
-               swiftUIScreenSizingOptions.contains(.preferredContentSize)
-            {
-                let size = view.sizeThatFits(view.frame.size)
-
-                if preferredContentSize != size {
-                    preferredContentSize = size
-                }
-            }
-        } else if swiftUIScreenSizingOptions.contains(.preferredContentSize) {
-            let size = view.sizeThatFits(view.frame.size)
+        if swiftUIScreenSizingOptions.contains(.preferredContentSize) {
```

Comment:
if this is the path we want to go, shall we extract the duplicate calculation logic?